src/App.js

```diff
@@ -11,33 +11,128 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      // game start
+      gameStart: false,
+      // game round
+      gameRound: 0,
+      // winner round
+      // set as null to make sure the class is empty, 0 is a value
+      roundWinner: null,
+      // players' current scores
+      player1CurrentScore: 0,
+      player2CurrentScore: 0,
+      // players' total scores
+      player1Total: 0,
+      player2Total: 0,
     };
   }
 
   dealCards = () => {
-    this.setState((state) => ({
+    // deal last 2 cards into newCurrCards
+    const newCurrCards = this.state.cardDeck.slice(-2);
+    if (newCurrCards[0].rank > newCurrCards[1].rank) {
+      // render states (player1CurrentScore and player1Total)
+      this.setState({
+        roundWinner: 1,
+        player1CurrentScore: this.state.player1CurrentScore + 1,
+        player1Total: this.state.player1Total + 1,
+      });
+    } else if (newCurrCards[1].rank > newCurrCards[0].rank) {
+      // render states (player2CurrentScore and player2Total)
+      this.setState({
+        roundWinner: 2,
+        player2CurrentScore: this.state.player2CurrentScore + 1,
+        player2Total: this.state.player2Total + 1,
+      });
+    } else {
+      // if no winner
+      this.setState({ roundWinner: 0 });
+    }
+
+    this.setState((prevState) => ({
       // Remove last 2 cards from cardDeck
-      cardDeck: state.cardDeck.slice(0, -2),
-      // Deal last 2 cards to currCards
-      currCards: state.cardDeck.slice(-2),
+      cardDeck: prevState.cardDeck.slice(0, -2),
+      // pop newCurrCards into currCards array
+      currCards: newCurrCards,
+      gameStart: true,
+      // update gameRound
+      gameRound: prevState.gameRound + 1,
     }));
   };
 
+  // reset game
+  resetGame = () => {
+    this.setState({
+      cardDeck: makeShuffledDeck(),
+      currCards: [],
+      hasGameStarted: false,
+      roundWinner: null,
+      player1Total: 0,
+      player2Total: 0,
+    });
```

Comment:
I think we could define a defaultState object or something along these lines, instead of rewriting our initial state